tsnet/tsnet.go

```diff
@@ -504,6 +504,91 @@ func (s *Server) APIClient() (*tailscale.Client, error) {
 	return c, nil
 }
 
+// ExposeHTTPS returns a HTTPS listener that is exposed over Funnel.
+// It will start the server if it has not been started yet.
+func (s *Server) ExposeHTTPS() (net.Listener, error) {
+	if err := s.Start(); err != nil {
+		return nil, err
+	}
+
+	ln, err := net.Listen("tcp", "[::1]:42069")
```

Comment:
??

Reply:
Please excuse us as this PR is very draft. I asked @Xe to post the PR as a draft so I could check it out and look into a peerapi issue. It's borked when using tsnet.